action.yml

```diff
@@ -1,28 +1,29 @@
 # See: https://docs.github.com/en/actions/creating-actions
-name: TODO:Action Name
+name: Gem Push
 author: FreeAgent
-description: TODO:Description
-#  inputs:
-#    foo:
-#      description: Foo level
-#      required: true
-#      default: 23
-#  outputs:
-#    bar:
-#      description: Bar output
+description: Push gem packages to a rubygems compatible repo
+inputs:
+  package-glob:
+    description: "File glob to match the .gem files to push"
+    default: "pkg/*.gem"
 
 runs:
-  # Pick one of the 3 possible types of action:
-
-  #using: 'composite'
-  #steps:
-  #  - name: Step One
-  #    shell: bash
-  #    run: |
-  #      echo TODO: Step One
-
-  #using: 'node12'
-  #main: main.js
-
-  #using: 'docker'
-  #image: 'Dockerfile'
+  using: "composite"
+  steps:
+    - name: Push Gem
+      shell: bash
+      env:
+        # Expects GEM_HOST and GEM_HOST_API_KEY to be set
```

Comment:
I really feel like if an action depends on something being set then it should expose that as an input. I've seen so many actions that need. a GitHub token require that ${{ secrets.GITHUB_TOKEN }} be passed explicitly (but I reckon they should default to it if it's not set).

Reply:
But that isn't the pattern you actually see in actions, when you use the ruby/setup action, it sets env vars. Do you think that is wrong and in fact the the ruby/setup should expose all the details of the ruby setup as outputs that you then have to explicitly pass to all the other ruby steps? This is modelled on the same thing. Docker login actions work the same way. Actions explicitly provide a mechanism to to export env for following atcions to use and encourage that.

The github token and secrets are a somewhat special case, they are not exposed to actions as context, unlike everything else, to stop leaking them and keep them hidden in logs and make it explicit what they use.

It also makes it more painfull when you want to use the same setup action outputs over lots of other actions. Feels like an awful lot of boiler plate, repetitious (error prone) code, that a good docs can cover.

Reply:
I can add input overrides if you want in a follow up.